OBJLoader: Add points support.

richardmonette opened the pull request:
<img width="291" alt="screen shot 2018-01-07 at 9 29 22 pm" src="https://user-images.githubusercontent.com/1041278/34657214-18935822-f3f2-11e7-8d3f-83cdcbc16f13.png">

Adds supports to the Obj loader for points.

```
p  v1 v2 v3 . . .

    Polygonal geometry statement.

    Specifies a point element and its vertex. You can specify multiple
    points with this statement. Although points cannot be shaded or
    rendered, they are used by other Advanced Visualizer programs.

    v is the vertex reference number for a point element. Each point
    element requires one vertex. Positive values indicate absolute
    vertex numbers. Negative values indicate relative vertex numbers.
```
http://paulbourke.net/dataformats/obj/

Fixes https://github.com/mrdoob/three.js/issues/13031  

@mrdoob

richardmonette committed:
add points to obj loader

makc commented:
how could you create the pull request just one hour ago and already have conflicts?

Mugen87 commented on examples/webgl_loader_obj.html:
```diff
@@ -93,7 +93,7 @@
 
 				var loader = new THREE.OBJLoader( manager );
-				loader.load( 'obj/male02/male02.obj', function ( object ) {
+				loader.load( 'obj/pointCloudBox/pointCloudBox.obj', function ( object ) {
```
I would not change the example. Just adding the `OBJ` file for testing should be sufficient for this feature.

Mugen87 commented on examples/js/loaders/OBJLoader.js:
```diff
@@ -658,13 +689,25 @@ THREE.OBJLoader = ( function () {
 					if ( ! material ) {
 
-						material = ( ! isLine ? new THREE.MeshPhongMaterial() : new THREE.LineBasicMaterial() );
+						if ( isLine ) {
```
Please ensure to follow the code style. You can check your code with https://zz85.github.io/mrdoobapproves/ 
This section should be look like this:
```javascript
if ( isLine ) {

   material = new THREE.LineBasicMaterial();

} else if ( isPoints ) {

   material = new THREE.PointsMaterial( { size: 1, sizeAttenuation: false } );

} else {

   material = new THREE.MeshPhongMaterial();

}

Mugen87 commented:
> how could you create the pull request just one hour ago and already have conflicts?

@richardmonette Have you created your PR on the latest `dev` branch?

richardmonette commented:
@Mugen87 Thank you for your notes, I will make the changes and rebase.

Mugen87 commented on examples/js/loaders/EXRLoader.js:
```diff
@@ -15,6 +15,485 @@ THREE.EXRLoader.prototype = Object.create( THREE.DataTextureLoader.prototype );
 
 THREE.EXRLoader.prototype._parser = function ( buffer ) {
 
+	const USHORT_RANGE = (1 << 16);
```
Why does changes to the `EXRLoader` popping up?

richardmonette replied:
Sorry about this, my git history has gotten a bit messed up, please ignore while I fix up the history.

Mugen87 commented on .gitignore:
```diff
@@ -7,3 +7,5 @@ node_modules
 .jshintrc
 .vs/
+*.sublime-project
+*.sublime-workspace
```
This should be no part of the PR.

richardmonette commented:
@Mugen87 Thank you for your patience while I was rebasing. I have cleared up the conflicts and addressed your comment about the formatting. I have also removed the changes to the example. If you'd like to test, this small patch tweaks the example so you can see the cube is loaded correctly.

```
diff --git a/examples/webgl_loader_obj.html b/examples/webgl_loader_obj.html
index 14e510e95..13886c0e8 100644
--- a/examples/webgl_loader_obj.html
+++ b/examples/webgl_loader_obj.html
@@ -55,7 +55,7 @@
                                document.body.appendChild( container );

                                camera = new THREE.PerspectiveCamera( 45, window.innerWidth / window.innerHeight, 1, 2000 );
-                               camera.position.z = 250;
+                               camera.position.z = 10;

                                // scene

@@ -93,7 +93,7 @@
                                };

                                var loader = new THREE.OBJLoader( manager );
-                               loader.load( 'obj/male02/male02.obj', function ( object ) {
+                               loader.load( 'obj/pointCloudBox/pointCloudBox.obj', function ( object ) {

                                        object.traverse( function ( child ) {

@@ -105,7 +105,7 @@

                                        } );

-                                       object.position.y = - 95;
+                                       //object.position.y = - 95;
                                        scene.add( object );

                                }, onProgress, onError );

mrdoob commented:
Thanks!
